src/schema/opportunity.ts

```diff
@@ -1061,44 +1093,57 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         isTeamMember: ctx.isTeamMember,
       });
 
-      return await queryPaginatedByDate<
-        GQLOpportunityMatch,
-        'updatedAt',
-        typeof args
-      >(
-        ctx,
-        info,
-        args,
-        { key: 'updatedAt', maxSize: 50 },
-        {
-          queryBuilder: (builder) => {
-            builder.queryBuilder
-              .where({ opportunityId: args.opportunityId })
-              .andWhere(`${builder.alias}.status IN (:...statuses)`, {
-                statuses: [
+      // If status is provided, filter by that status; otherwise use all allowed statuses
+      const statusesToFilter = args.status ? [args.status] : allowedStatuses;
+
+      const [connection, totalCount] = await Promise.all([
+        queryPaginatedByDate<GQLOpportunityMatch, 'updatedAt', typeof args>(
```

Comment:
Is this now used for tinder style matching? Do we want to make sure user can't browse multiple candidates if doing direct API requests and requesting more candidates, and only do one by one? (could be something we add later, just for shenanigans protection) 😆

Reply:
Where do we show it as list? (just for my own reference)